configstack/module.go

```diff
@@ -650,3 +655,62 @@ func buildDirList(terragruntOptions *options.TerragruntOptions, topLevelDir stri
 	}
 	return pathsToCheck, nil
 }
+
+// ListStackDependentModules - build a map with each module and its dependent modules
+func ListStackDependentModules(stack *Stack) map[string][]string {
+	// build map of dependent modules
+	// module path -> list of dependent modules
+	var dependentModules = make(map[string][]string)
+
+	// build initial mapping of dependent modules
+	for _, module := range stack.Modules {
+
+		if len(module.Dependencies) != 0 {
+			for _, dep := range module.Dependencies {
+				dependentModules[dep.Path] = util.RemoveDuplicatesFromList(append(dependentModules[dep.Path], module.Path))
+			}
+		}
+	}
+
+	// Floyd–Warshall inspired approach to find dependent modules
+	// merge map slices by key until no more updates are possible
```

Comment:
Ah, much clearer, thank you!

Reply:
One NIT: we should probably have unit tests for this method in particular, testing it out against all sorts of graphs, including those with cycles. But that can be added in a follow up PR, no need to hold up this one.

Reply:
Good point, added tests